lib/routing/router.ts

```diff
@@ -1,50 +1,69 @@
-import { Routes, Route } from './routes';
+import { Route } from './routes';
 import { Middleware } from '../middleware/middleware';
 import { Controller } from '../controller/controller';
 import { GenericConstructor } from '../interfaces';
 import { Request } from '../request/request';
 import { Response } from '../response';
 import { Container } from 'inversify';
+import { HttpMethods } from '../request/event';
 
 
 export class Router<M extends Middleware, C extends Controller, R extends Route<M, C>> {
 
   private middleware: GenericConstructor<M>[] = [];
-  private subjectRoute: Route<M, C>;
+  private subjectRoute: R; 
+  private requestRoute: string;
+  private requestMethod: HttpMethods;
+  private pathParams: {};
 
   constructor(
     private request: Request,
     private container: Container
   ) { }
 
-  public route(routes: Routes<M, C, R>): void {
-    let requestRoute: string = this.request.getResource()
-    let requestMethod: string = this.request.getRequestMethod();
-    let routeGroup = routes.routes[requestRoute];
+  public route(routes: R[]): void {
+    this.requestRoute = this.request.getResource()
+    this.requestMethod = this.request.getRequestMethod();
 
-    this.addMiddlewareIfExists(routes.middleware);
```

Comment:
Also what happened to these middlewares? Just axing them?

Reply:
take a look how routes are definied in the middleware, there are no groups anymore. I mentioned this breifly on our call. In the future we will want to make like route groups and shit but fine for now. Could be an extension even